src/System.CommandLine/src/System/CommandLine/ArgumentSyntax.cs

```diff
@@ -52,12 +53,7 @@ private void Validate()
                 var helpText = GetHelpText();
                 Console.Error.Write(helpText);
 
-                // TODO: This should use Environment.Exit(0) but this API isn't available yet.
```

Comment:
All right. Makes sense. We can make sure we are depending on a particular version of commandline and only depend on the new once once we change our code to handle the help requested properly.